### PR TITLE
test: Add Kubernetes Service CI tests for IPv6

### PR DESCRIPTION
**Pull request by gandro**
This adds initial connectivity tests for IPv6 in the Kubernetes service test suite. The deployed Kubernetes is not yet running in dual-stack mode - therefore we install the Cilium service rules manually via CLI. Because of this, the IPv6 tests are only ran in kube-proxy-free mode, as kube-proxy only installs IPv4 rules.

NodePort and ExternalIP tests have yet to be implemented in a follow-up PR. This PR here lays the groundwork by upgrading all images to listen on IPv6, providing helpers to install services via Cilium CLI, and adding basic ClusterIP tests.

Ref #9362 

This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10115)

### CODE REVIEW

**Comment by gandro** [comment]
test-focus K8sService*

**Comment by gandro** [comment]
test-focus K8sService*

**Comment by gandro** [comment]
test-focus K8sService*

**Comment by gandro** [comment]
test-focus K8sService*

**Comment by gandro** [comment]
test-focus K8sService*

**Comment by gandro** [comment]
test-focus K8sService*

**Comment by gandro** [comment]
test-me-please

File: test/k8sT/Services.go [hunk 1]
```diff
@@ -104,6 +104,37 @@ var _ = Describe("K8sServicesTest", func() {
...
+			Expect(err).To(BeNil(), "Failed to add cilium service")
+		}
+	}
+
+	ciliumDelService := func(id int64) {
+		ciliumPods, err := kubectl.GetCiliumPods(helpers.CiliumNamespace)
+		Expect(err).To(BeNil(), "Cannot get cilium pods")
+		for _, pod := range ciliumPods {
+			// ignore result so tear down still continues on failures
+			_ = kubectl.CiliumServiceDel(pod, 1001)
```
Comment (brb) [inline test/k8sT/Services.go]: s/1001/id/?

**Review by brb — CHANGES_REQUESTED** [review]
LGTM, just one nit.

File: test/k8sT/Services.go (hunk 1 above)
Comment (gandro) [reply test/k8sT/Services.go]: :scream: Thanks, good catch!

**Comment by gandro** [comment]
test-me-please

**Review by brb — APPROVED** [review]

**Pull request merged**